docs/developer-docs/getting-started/write-smart-contracts.mdx

```diff
@@ -0,0 +1,560 @@
+      };
+  };
+
+  // Use a query call to get the current random number.
+  public query func getCurrentNumber() : async Nat {
+      currentRandomNumber
+  };
+
+  // Initialize a timer to generate a new number every 5 seconds.
+  let timer = Timer.recurringTimer<system>(#seconds 5, generateNewNumber);
+
+  // highlight-start
+
+  // Define a public function to get the daily stats about ICP from the ICP API.
+  public func getIcpInfo() : async Text {
+
+    // Define the API endpoints to obtain data from.
+    // This example uses the IC API, but any API endpoint can be used.
+    let url = "https://ic-api.internetcomputer.org/api/v3/daily-stats?format=json";
+    let transform_context : Types.TransformContext = {
+      function = transform;
+      context = Blob.fromArray([]);
+    };
+
+    // Define the http_request components.
+    let http_request = {
+        url = url;
+        max_response_bytes = null; // Optional.
+        headers = [];
+        body = null; // Optional.
+        method = #get;
+        transform = ?transform_context;
+    };
+
+    // HTTP outcalls require cycles to be attached to the call.
+    Cycles.add<system>(20_949_972_000);
+
+    // Execute the HTTP outcall.
+    let http_response = await IC.http_request(http_request);
+
+    let response_body: Blob = http_response.body;
+    switch (Text.decodeUtf8(response_body)) {
+        case null { "No value returned" };
+        case (?y) { y };
+    };
+  };
+
+  // Define a transform function to return the status response and body.
+  public query func transform(raw : Types.TransformArgs) : async IC.http_request_result {
+      {
+          status = raw.response.status;
+          body = raw.response.body;
+          headers = [ ];
+      };
+  };
+// highlight-end
+}
+```
+
+This HTTP outcalls code references custom types, as indicated by the import statement `import Types "Types";` and code statements like `Types.TransformContext`. Custom type definitions can be created from within the `main.mo` file, but for types that may be used in several different source code files, it can be beneficial to define them in a single `Types.mo` file, and then import them as needed.
+
+Create a new file called `src/PROJECT_NAME_backend/Types.mo` that contains the following:
+
+```motoko title="src/PROJECT_NAME_backend/Types.mo"
+import IC "ic:aaaaa-aa";
+
+module Types {
+
+  // HTTPS outcalls have an optional "transform" key. These two types help describe it.
+  // The transform function may transform the body in any way, add or remove headers, modify headers, etc.
+
+  public type TransformArgs = {
+    response : IC.http_request_result;
+    context : Blob;
+  };
+
+  public type TransformContext = {
+    function : shared query TransformArgs -> async IC.http_request_result;
+    context : Blob;
+  };
+
+}
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust title="src/PROJECT_NAME_backend/src/lib.rs"
+
+
+```
+</TabItem>
+</AdornedTabs>
+
```

Comment:
@jessiemongeon1 here we should guide the user to the candid ui and/or provide a command to locally call `getIcpInfo` otherwise this addition is not really being excercised